[sentry-native] build failure / problem with curl dependency

**Host Environment**
- OS: Arch linux
- Compiler: clang 11.0.0

**To Reproduce**
Steps to reproduce the behavior:
1. Create a CMake project depending on sentry-native
2. Use `target_link_libraries(EXAMPLE_APP_NAME sentry::sentry)`

**Failure logs** 
```
CMake Error at cmake-build-debug/vcpkg_installed/x64-linux/share/sentry/sentry-config.cmake:42 (set_property):
  Property INTERFACE_LINK_LIBRARIES may not contain link-type keyword
  "optimized".  The INTERFACE_LINK_LIBRARIES property may contain
  configuration-sensitive generator-expressions which may be used to specify
  per-configuration rules.
```

**Additional context**
Content of generated file `sentry-config.cmake`:
```cmake
####### Expanded from @PACKAGE_INIT@ by configure_package_config_file() #######
####### Any changes to this file will be overwritten by the next CMake run ####
####### The input file was sentry-config.cmake.in                            ########

get_filename_component(PACKAGE_PREFIX_DIR "${CMAKE_CURRENT_LIST_DIR}/../../" ABSOLUTE)

macro(set_and_check _var _file)
  set(${_var} "${_file}")
  if(NOT EXISTS "${_file}")
    message(FATAL_ERROR "File or directory ${_file} referenced by variable ${_var} does not exist !")
  endif()
endmacro()

macro(check_required_components _NAME)
  foreach(comp ${${_NAME}_FIND_COMPONENTS})
    if(NOT ${_NAME}_${comp}_FOUND)
      if(${_NAME}_FIND_REQUIRED_${comp})
        set(${_NAME}_FOUND FALSE)
      endif()
    endif()
  endforeach()
endmacro()

####################################################################################

set(SENTRY_BACKEND inproc)
set(SENTRY_TRANSPORT curl)

if(SENTRY_BACKEND STREQUAL "crashpad")
	if()
		find_package(crashpad REQUIRED)
	else()
		include("${CMAKE_CURRENT_LIST_DIR}/sentry_crashpad-targets.cmake")
	endif()
endif()

include("${CMAKE_CURRENT_LIST_DIR}/sentry-targets.cmake")

if(SENTRY_TRANSPORT STREQUAL "curl" AND NOT OFF)
	find_package(CURL REQUIRED)
	set_property(TARGET sentry::sentry APPEND
		PROPERTY INTERFACE_LINK_LIBRARIES ${CURL_LIBRARIES})
endif()
```

--- a/ports/sentry-native/CONTROL
+++ b/ports/sentry-native/CONTROL
@@ -1,5 +1,6 @@
 Source: sentry-native
 Version: 0.4.4
+Port-Version: 1
 Homepage: https://sentry.io/
 Description: Sentry SDK for C, C++ and native applications.
 Build-Depends: curl (!windows)

--- a/ports/sentry-native/fix-config-cmake.patch
+++ b/ports/sentry-native/fix-config-cmake.patch
@@ -1,0 +1,23 @@
+diff --git a/sentry-config.cmake.in b/sentry-config.cmake.in
+index 89ea345..21d8732 100644
+--- a/sentry-config.cmake.in
++++ b/sentry-config.cmake.in
+@@ -5,7 +5,7 @@ set(SENTRY_TRANSPORT @SENTRY_TRANSPORT@)
+ 
+ if(SENTRY_BACKEND STREQUAL "crashpad")
+ 	if(@SENTRY_CRASHPAD_SYSTEM@)
+-		find_package(crashpad REQUIRED)
++		find_dependency(crashpad)
+ 	else()
+ 		include("${CMAKE_CURRENT_LIST_DIR}/sentry_crashpad-targets.cmake")
+ 	endif()
+@@ -14,7 +14,7 @@ endif()
+ include("${CMAKE_CURRENT_LIST_DIR}/sentry-targets.cmake")
+ 
+ if(SENTRY_TRANSPORT STREQUAL "curl" AND NOT @BUILD_SHARED_LIBS@)
+-	find_package(CURL REQUIRED)
++	find_dependency(CURL)
+ 	set_property(TARGET sentry::sentry APPEND
+-		PROPERTY INTERFACE_LINK_LIBRARIES ${CURL_LIBRARIES})
++		PROPERTY INTERFACE_LINK_LIBRARIES CURL::libcurl)
+ endif()

--- a/ports/sentry-native/portfile.cmake
+++ b/ports/sentry-native/portfile.cmake
@@ -9,8 +9,10 @@ vcpkg_download_distfile(ARCHIVE
 vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
     ARCHIVE ${ARCHIVE}
-    PATCHES fix-warningC5105.patch
     NO_REMOVE_ONE_LEVEL
+    PATCHES
+        fix-warningC5105.patch
+        fix-config-cmake.patch
 )
 
 if (NOT DEFINED SENTRY_BACKEND)

